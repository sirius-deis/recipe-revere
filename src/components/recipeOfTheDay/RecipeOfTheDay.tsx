import { FC } from "react";
import { useQuery } from "@apollo/client";
import Loader from "../loader/Loader";
import RecipePreview from "../recipePreview/RecipePreview";
import { RECIPE_OF_THE_DAY } from "../../queries/queries";
import ErrorBox from "../errorBox/ErrorBox";

const RecipeOfTheDay: FC = () => {
  const { loading, error, data } = useQuery(RECIPE_OF_THE_DAY, {
    variables: {
      tags: "RECIPE_OF_THE_DAY",
    },
  });
  if (loading) {
    return <Loader />;
  }
  if (error) {
    return <ErrorBox message={error.message} />;
  }
  if (!data) {
    return <div>Something went wrong</div>
  }
  const { getRecipe } = data;
  return (
    <>
      <RecipePreview
        recipe={{
          id: getRecipe?.recipe.url, label: getRecipe?.recipe.label,
          image: getRecipe?.recipe.image, dietLabels: getRecipe?.recipe.dietLabels
        }}

        avgRating={getRecipe?.averageRating}
        btnTitle="Cook now"
      />
    </>
  );
};

export default RecipeOfTheDay;
