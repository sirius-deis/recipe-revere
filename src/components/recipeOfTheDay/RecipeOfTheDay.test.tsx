import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import RecipeOfTheDay from "./RecipeOfTheDay";

jest.mock("@apollo/client", () => ({
  ...jest.requireActual("@apollo/client"),
  useQuery: jest.fn(),
}))

import { useQuery } from "@apollo/client";

jest.mock("../loader/Loader", () => () => <div>Loading</div>)

describe("RecipeOfTheDay", () => {
  const mockData = {
    getRecipe: {
      recipe: {
        url: "1",
        label: "Recipe of the Day",
        image: "img1.jpg",
        dietLabels: ["Vegan"],
      },
      averageRating: 4.5,
    },
  };
  it("should match snapshot", () => {
    (useQuery as jest.Mock).mockReturnValue({
      loading: false,
      error: null,
      data: mockData,
    });
    const { container } = render(<Router><RecipeOfTheDay /></Router>);
    expect(container).toMatchSnapshot();
  })
  it("Should render loading state", () => {
    (useQuery as jest.Mock).mockReturnValue({
      loading: true
    });
    render(<Router><RecipeOfTheDay /></Router>);
    expect(screen.getByText("Loading")).toBeInTheDocument();
  })
  it("Should render error state", () => {
    const errorMessage = 'Error fetching recipe of the day';
    (useQuery as jest.Mock).mockReturnValue({
      loading: false,
      error: new Error(errorMessage)
    });
    render(<Router><RecipeOfTheDay /></Router>)
    expect(screen.queryByText(errorMessage)).toBeInTheDocument();
  });
  it("Should render error message if data is null", () => {
    (useQuery as jest.Mock).mockReturnValue({
      loading: false,
      error: null,
      data: null,
    });
    render(<Router><RecipeOfTheDay /></Router>);
    expect(screen.getByText("Something went wrong")).toBeInTheDocument();
  })
})