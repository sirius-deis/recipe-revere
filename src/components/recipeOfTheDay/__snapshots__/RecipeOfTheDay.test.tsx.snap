// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeOfTheDay should match snapshot 1`] = `
<div>
  <div
    class="panel bordered shadowed vertical"
    data-testid="panel"
  >
    <div
      class="recipe-container"
    >
      <div
        class="image-container"
        data-testid="img"
        style="background-image: url(img1.jpg);"
      >
        <div
          class="rating"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          4.5
        </div>
        <div
          class="liked"
        >
          <button
            class="btn btn-sm btn-icon btn-rounded"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="desc"
      >
        <div
          class="row"
          style="justify-content: space-between; gap: 0.7rem;"
        >
          <div
            class="col"
            style="gap: 0.5rem; max-width: 65%;"
          >
            <h3
              class="title"
            >
              Recipe of the Day
            </h3>
            <ul
              class="tags"
            >
              <li>
                Vegan
              </li>
            </ul>
          </div>
          <div
            class="col"
          >
            <a
              href="/recipe/1"
            >
              <button
                class="btn btn-sm btn-action"
              >
                Cook now
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
